pkgs/applications/networking/browsers/firefox/common.nix

```diff
@@ -322,6 +321,42 @@ buildStdenv.mkDerivation ({
     gappsWrapperArgs+=(--argv0 "$out/bin/.${binaryName}-wrapped")
   '';
 
+  # FIXME: The separateDebugInfo hook skips artifacts whose Build ID's length is not 40.
+  # But we got 16-length Build ID here. Override it to make it work.
+  # Remove it when PR #146275 is merged.
+  preFixup = lib.optionalString enableDebugSymbols ''
+    _separateDebugInfo() {
+        [ -e "$prefix" ] || return 0
+
+        local dst="''${debug:-$out}"
+        if [ "$prefix" = "$dst" ]; then return 0; fi
+
+        dst="$dst/lib/debug/.build-id"
+
+        # Find executables and dynamic libraries.
+        local i
+        while IFS= read -r -d $'\0' i; do
+            if ! isELF "$i"; then continue; fi
+
+            # Extract the Build ID. FIXME: there's probably a cleaner way.
```

Comment:
```suggestion
            # Extract the build ID.
            # FIXME: There's probably a cleaner way.
```

Reply:
This is copied from current separateDebugInfo hook code. I'd not change it if not necessary.